### ISSUE TITLE
Cannot open accounts when I customize fields

### ISSUE THREAD

**Issue by Wayne-II** [salesagility/SuiteCRM#6777]
#### Issue
When I remove "required" from the name field in the Accounts module, this poses a problem.  If I don't enter a name, there is no links in the accounts list to click in order to access the record.

I removed the name field because I wanted it to be split up into first name and last name.

#### Expected Behavior
I should be able to click something other than the name in order to access a record, especially when the required parameter of name is removed.

#### Actual Behavior
If I remove 

#### Possible Fix
1) New field parameter: Click Action
2) Make any non-inline editable field clickable to access the account
3) Allow user to configure which field is the click field in order to access the account.

#### Steps to Reproduce
1) Fresh install
2) Admin -> Studio -> Accounts -> Fields -> name( Company )
3) Remove the required checkbox.
4) Create an account without a name.  
5) Try to access the record.

#### Context
The entire system is unusable

#### Your Environment

* SuiteCRM Version used: 7.11
* Browser name and version Chrome, IE, and Edge
* WAMP with various version of PHP test 5.6.x, 7.0.x, 7.1.x, 7.2.x and 7.3.x
* WIndows 7, 10, and XP

**Comment by Wayne-II** [comment]
In case someone comes across this, after a couple hours in the code, I found listviewdefs.php under modules/Accounts/metadata/listviewdefs.php.  There is an interestion section of code which has 'link' set to true for 'name'.  I suspect if I find how I can customize listviewdefs I can set link to true for any of my custom fields.

**Comment by Wayne-II** [comment]
I did a search for listviewdefs.php in the suitcrm source directory and found an if checking for custom/modules/Accounts/metadata/listviewvardefs.php.

That path is from memory, but setting link to the account # field resolved my issue.  I suspect I will find the same type of configuration for the "search view" so that I can add account id to search view.

Hope this helps someone.  I spent 2 days on this already.  I was ready to tell them to use the quick edit button to view accounts until I found the search view issue.

P.S. Not sure it's called "search view", that's just what I'm calling it until I know better.

P.P.S. I did find a lot of people in issues and forums running into what I suspect is this issue when customizing their fields for various modules.  The usual suggested steps to fix was "remove customizations".  One person had suggested hiring a dev to run through the code in a debugger, but there were no errors in the logs.  That OP ended up getting bitter.

**Comment by Wayne-II** [comment]
You cannot access the record if you search for it, there is no "edit" pen icon in the search results.

**Comment by Wayne-II** [comment]
That's a bit unclean after reading it.

I didn't notice the "View" Link once you had clicked the "edit" icon, and just telling them to use the edit view was sufficient if it was the only issue.  The issue remains, you still cannot access a nameless record from search results.

**Comment by Wayne-II** [comment]
Doing a quick search, my hopes were dashed by these two lines of code:

https://github.com/salesagility/SuiteCRM/blob/master/modules/Home/UnifiedSearch.php#L111

https://github.com/salesagility/SuiteCRM/blob/master/modules/Home/UnifiedSearch.php#L144

It is unlikely my employer will provide me with sufficient time to fix these in a manner which would result in a successful pull request, and my at home time is currently in deficit.  I can make this configurable at some point, in the non-near feature.

**Comment by Wayne-II** [comment]
You are correct, the edit button does appear when using search in that demo.  I will upload screenshots with info redacted due to it's production nature.

**Comment by Wayne-II** [comment]
![nosearchedit](https://user-images.githubusercontent.com/42722399/51394527-f710a280-1b10-11e9-927d-a6ec1e782084.png)

The search renders modules/Home/UnifiedSearch.php.  Nothing I can do allows it to use UnifiedSearchAdvanced.php, which seems like it would use my custom listviewvardefs.php.

**Comment by Wayne-II** [comment]
![nosearchedit2](https://user-images.githubusercontent.com/42722399/51394749-88801480-1b11-11e9-9832-8188437bba39.png)

I'm just using the magnifying glass in the top right corner of the page.

**Comment by Wayne-II** [comment]
I appreciate the assistance troubleshooting.  So far I'm very please with this software.  Bit of a learning curve, but once you know how things are done, and a bit of PHP, it's not too hard.

Any chance you found whereabouts the issue lies so I can at least get it up and running?

**Comment by Dillon-Brown** [comment]
@Wayne-II Unfortunately I don't have the AOD stuff setup atm but i'm guessing it's probably in modules/Home/UnifiedSearch.php

**Comment by Wayne-II** [comment]
In case anyone runs into this bug, I found a very hacky workaround.

I changed line https://github.com/salesagility/SuiteCRM/blob/master/modules/Home/UnifiedSearch.php#L158
`function getRecordSummary(SugarBean $bean){`
to
`function getRecordSummary(SugarBean $bean, $record_id){`

I changed line https://github.com/salesagility/SuiteCRM/blob/master/modules/Home/UnifiedSearch.php#L249
`$newHit->summary = getRecordSummary($bean);`
to
`$newHit->summary = getRecordSummary($bean, $hit->record_id);`

and I changed line https://github.com/salesagility/SuiteCRM/blob/master/modules/Home/UnifiedSearch.php#L180
`$summary[] = $bean->$key;`
to 
`if( $key == 'account_number_c'){
    $link = 'http://192.168.10.140:81/index.php?module=Accounts&action=DetailView&record=' . $record_id;
    $summary[] = '<a href="' . $link . '"/>' . $bean->$key . '</a>';
}else{
    $summary[] = $bean->$key;
}`
This hack is specific to my implementation.  It could be better but this suits my needs for now.
You would change $key == 'account_number_c' to whatever custom field that shows up in your summary results.  You can find that name in the studio fields customization area.  It will be lower case as the code strToLower it for comparrison.

Excuse the poor formatting.  Not sure what GitHub does but it looks funny in preview.
